pydicom/tests/test_filewriter.py

```diff
@@ -1725,6 +1791,94 @@ def test_meta_unchanged(self):
+        fp.is_little_endian = True
+        elem = DataElement(0x00100010, 'US', '')
+        fmt = 'H'
+        write_numbers(fp, elem, fmt)
+        assert fp.getvalue() == b''
+
+    def test_write_list(self):
+        """Test writing an element value with VM > 1"""
+        fp = DicomBytesIO()
+        fp.is_little_endian = True
+        elem = DataElement(0x00100010, 'US', [1, 2, 3, 4])
+        fmt = 'H'
+        write_numbers(fp, elem, fmt)
+        assert fp.getvalue() == b'\x01\x00\x02\x00\x03\x00\x04\x00'
+
+    def test_write_singleton(self):
+        """Test writing an element value with VM = 1"""
+        fp = DicomBytesIO()
+        fp.is_little_endian = True
+        elem = DataElement(0x00100010, 'US', 1)
+        fmt = 'H'
+        write_numbers(fp, elem, fmt)
+        assert fp.getvalue() == b'\x01\x00'
+
+    def test_exception(self):
+        """Test exceptions raise IOError"""
+        fp = DicomBytesIO()
+        fp.is_little_endian = True
+        elem = DataElement(0x00100010, 'US', b'\x00')
+        fmt = 'H'
+        assert_raises_regex(IOError,
+                            "for data_element:\n\(0010, 0010\)",
+                            write_numbers,
+                            fp,
+                            elem,
+                            fmt)
+
+    def test_write_big_endian(self):
+        """Test writing big endian"""
+        fp = DicomBytesIO()
+        fp.is_little_endian = False
+        elem = DataElement(0x00100010, 'US', 1)
+        fmt = 'H'
+        write_numbers(fp, elem, fmt)
+        assert fp.getvalue() == b'\x00\x01'
+
+
+class TestWritePN(object):
+    """Test filewriter.write_PN"""
+    @pytest.mark.skip("Raises exception due to issue #489")
+    def test_no_encoding_unicode(self):
+        """If PN element as no encoding info, default is used"""
+        fp = DicomBytesIO()
+        fp.is_little_endian = True
+        elem = DataElement(0x00100010, 'PN', u'\u03b8')
+        write_PN(fp, elem)
+
+    def test_no_encoding(self):
+        """If PN element as no encoding info, default is used"""
+        fp = DicomBytesIO()
+        fp.is_little_endian = True
+        elem = DataElement(0x00100010, 'PN', 'Test')
+        write_PN(fp, elem)
+        assert fp.getvalue() == b'Test'
+
+
+class TestWriteDT(object):
+    """Test filewriter.write_DT"""
+    def test_format_dt(self):
+        """Test _format_DT"""
+        elem = DataElement(0x00181078, 'DT', DT('20010203123456.123456'))
+        assert hasattr(elem.value, 'original_string')
+        assert _format_DT(elem.value) == '20010203123456.123456'
+        del elem.value.original_string
+        assert not hasattr(elem.value, 'original_string')
+        assert elem.value.microsecond > 0
+        assert _format_DT(elem.value) == '20010203123456.123456'
+
+        elem = DataElement(0x00181078, 'DT', DT('20010203123456'))
+        del elem.value.original_string
+        assert _format_DT(elem.value) == '20010203123456'
+
```

Comment:
generally, I would test only the public functions (in this case `write_DT`) without relying on private implementations